### PR TITLE
Port CoreFx PR #10868 Full-version package dependency verification.

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
* Moves package validation out of dir.props
* Leverages dotnet/versions repo for updating and validating package versions.

@dagood could you confirm this looks good? I think commented-out sections in dependencies.props can be removed, is there anything else you think I can remove from our version?

### CODE REVIEW

File: dependencies.props [hunk 1]
```diff
@@ -0,0 +1,92 @@
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!-- Source of truth for dependency tooling: the commit hash of the dotnet/versions master branch as of the last auto-upgrade. -->
+  <PropertyGroup>
+    <CoreFxCurrentRef>996189e56e6cc7e6f71a371e18efde7d4f8e104f</CoreFxCurrentRef>
+    <CoreClrCurrentRef>5ceb13b6c4b9b6b95c0b77b59c0477475e7bba54</CoreClrCurrentRef>
+    <ExternalCurrentRef>cdb8679634e573847e20c1eae0fa3918f5daf1e6</ExternalCurrentRef>
```
Comment (StephenBonikowsky) [inline dependencies.props]: Forgot to remove this.

**Comment by iamjasonp** [comment]
Question: does this need to also go into the ws-trust branch?

**Comment by roncain** [comment]
LGTM.
Once this is in, I think I can just cherry-pick the commit over to the ws-trust branch.  I may wait till PR https://github.com/dotnet/dotnet-ci/pull/466 has been merged so we can double check it runs in CI in that branch.

File: dependencies.props [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    <WcfCurrentRef>8373288e4fc75df9f37e54f8d6bab15d0e07a083</WcfCurrentRef>
+  </PropertyGroup>
+
+  <!-- Auto-upgraded properties for each build info dependency. -->
+  <PropertyGroup>
+    <CoreFxExpectedPrerelease>beta-24512-01</CoreFxExpectedPrerelease>
+    <CoreClrExpectedPrerelease>beta-24512-03</CoreClrExpectedPrerelease>
+    <WCFExpectedPrerelease>beta-24431-01</WCFExpectedPrerelease>
```
Comment (dagood) [inline dependencies.props]: From just searching in github it looks like only `CoreFxExpectedPrerelease` is used--if that's true, you can remove the other two. (And their `XmlUpdateStep`s.)

File: dependencies.props [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  </ItemGroup>
+
+  <!-- Set up dependencies on packages that aren't found in a BuildInfo. -->
+  <ItemGroup>
+    <!--<TargetingPackDependency Include="Microsoft.TargetingPack.NETFramework.v4.5" />
+    <TargetingPackDependency Include="Microsoft.TargetingPack.NETFramework.v4.5.1" />
+    <TargetingPackDependency Include="Microsoft.TargetingPack.NETFramework.v4.5.2" />-->
+    <TargetingPackDependency Include="Microsoft.TargetingPack.NETFramework.v4.6" />
+    <!--<TargetingPackDependency Include="Microsoft.TargetingPack.NETFramework.v4.6.1" />
+    <TargetingPackDependency Include="Microsoft.TargetingPack.NETFramework.v4.6.2" />-->
```
Comment (dagood) [inline dependencies.props]: Agreed, the commented out dependencies in this ItemGroup can be removed.

File: build.proj [hunk 4]
```diff
@@ -108,23 +107,6 @@
...
-                                       ValidationPatterns="@(ValidationPattern)" />
-  </Target>
-
-  <Target Name="UpdateInvalidPackageVersions">
-    <ValidateProjectDependencyVersions ProjectJsons="@(ProjectJsonFiles)"
-                                       ProhibitFloatingDependencies="$(ProhibitFloatingDependencies)"
-                                       ValidationPatterns="@(ValidationPattern)"
-                                       UpdateInvalidDependencies="true" />
-  </Target>
-
```
Comment (dagood) [inline build.proj]: I think you should also remove `UpdateDependencies.ps1` since it won't work anymore.

File: dependencies.props (hunk 2 above)
Comment (StephenBonikowsky) [inline dependencies.props]: I'm seeing that we depend on 3 CoreClr packages...
"Microsoft.NETCore.TestHost" - in Common\test-runtime\project.json
"Microsoft.NETCore.Runtime.CoreCLR" - in Common\test-runtime\project.json
"Microsoft.TargetingPack.Private.CoreCLR" - in each of our ServiceModel contracts src\netcore50\project.json

Comment (dagood) [inline dependencies.props]: Oh, these properties are only for msbuild files to use--project.jsons will be upgraded regardless as long as the `RemoteDependencyBuildInfo` is there.

File: build.proj (hunk 4 above)
Comment (StephenBonikowsky) [inline build.proj]: Will do.

File: dependencies.props (hunk 2 above)
Comment (StephenBonikowsky) [inline dependencies.props]: Ah, I see, thought they were still being used in the same way as before.

**Pull request merged**